### PR TITLE
Make both `/ui/?query=` and `/ui?query=` work

### PR DESCRIPTION
**Pull request by hi-rustin**
Follow up https://github.com/grafana/phlare/pull/645

Test locally:
- Before change
image

- After change
image

### CODE REVIEW

**Review by simonswine — COMMENTED** [review]
I would prefer to redirect `/ui` to `/ui/`

**Comment by hi-rustin** [comment]
> I would prefer to redirect `/ui` to `/ui/`

Usually, we use `/ui?a=xxx` instead of `/ui/?a=xxx`. For example: `https://github.com/issues?q=is%3Aopen+is%3Aissue+author%3Ahi-rustin+archived%3Afalse+phlare`.

I think adding a redirection here is a little bit weird. It is not common to have this kind of redirection.

**Comment by hi-rustin** [comment]
> Honestly I didn't do any work here hoping that we would default the UI to the root instead of `/ui/` once most pages were ready. But oh well.

Off-topic: I would like to ask what is the relationship between this UI and the UI in grafana? Will they be merged? And what is the relationship with pyroscope UI?

**Comment by Rperry2174** [comment]
> Off-topic: I would like to ask what is the relationship between this UI and the UI in grafana? Will they be merged? And what is the relationship with pyroscope UI?

We will provide more official guidance in this at our next community meeting / via other channels. 

However, just so you are in the loop as you've been helping us with this (thanks btw!) our long term goal is merging the Pyroscope and Phlare projects. 

Once they are merged there will likely be two ways to view profiling data:
1. Via standalone UI (the one being talked about here)
2. Via Grafana UI (explore tab and deeper integrations into Grafana)

In the meantime, we've made this UI compatible with both phlare and pyroscope projects to unify the frontend which users interact with while we work on merging the backend. 

That way when the projects are eventually merged it will be done in the "background" with minimum disruption to users (from a UI perspective)

**Comment by hi-rustin** [comment]
> Once they are merged there will likely be two ways to view profiling data:
> 
> 1. Via standalone UI (the one being talked about here)
> 2. Via Grafana UI (explore tab and deeper integrations into Grafana)

Sounds great! Thanks for your reply. I'm not confused at all now.

**Comment by hi-rustin** [comment]
> I don't think it is weird, usually this redirection is done to support relative linking of assets and APIs.
> 
> Maybe an example to make it clearer:
> 
> * User wants Phlare to mount a via a reverse proxy in there `https://tooling.corp/phlare`
> * When you are accessing it using `/phlare/ui`, the API will be under `../api`
> * When you are accessing it using `/phlare/ui/`, the API will be under `../../api`.

Sounds reasonable. Thanks for your review!



> Honestly I didn't do any work here hoping that we would default the UI to the root instead of `/ui/` once most pages were ready. But oh well.

Make sense. I also prefer to move UI to the root and move admin pages to `/admin` or other routes. It makes accessing the data easier.

File: pkg/api/api.go [hunk 1]
```diff
@@ -147,6 +147,9 @@ func (a *API) RegisterAPI(statusService statusv1.StatusServiceServer) error {
 	a.RegisterRoutesWithPrefix("/ui/assets/", http.StripPrefix("/ui/", http.FileServer(uiAssets)), false, true, "GET")
 	// Serve index to all other pages
 	a.RegisterRoutesWithPrefix("/ui/", uiIndexHandler, false, true, "GET")
+	// Redirect `/ui` to `/ui/`.
+	// See more: https://github.com/grafana/phlare/pull/649#issuecomment-1522958157.
+	a.RegisterRoute("/ui", http.RedirectHandler("/ui/", http.StatusMovedPermanently), false, true, "GET")
```
Comment (simonswine) [inline pkg/api/api.go:152]: Nit: I think a 302 redirect has less risk, should we choose to do something else in the future

 ```suggestion
	a.RegisterRoute("/ui", http.RedirectHandler("/ui/", http.StatusFound), false, true, "GET")
```

Comment (simonswine) [inline pkg/api/api.go:152]: Nit: I think a 302 redirect has less risk, should we choose to do something else in the future

```suggestion
	a.RegisterRoute("/ui", http.RedirectHandler("/ui/", http.StatusFound), false, true, "GET")
```

**Review by simonswine — APPROVED** [review]
LGTM

File: pkg/api/api.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/api/api.go]: Thanks! Make sense.

**Pull request merged**